### PR TITLE
refactor: optimize and fix

### PR DESCRIPTION
**Pull request by devhaozi**
Closes # 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: rotatelogs_test.go
Comment (hwbrzzl) [inline rotatelogs_test.go:1]: Q: Why remove these cases?

Comment (devhaozi) [reply rotatelogs_test.go:1]: > Q: Why remove these cases?

They are so many failed cases, doesn't know why, maybe it doesn't design for Windows.

Comment (hwbrzzl) [reply rotatelogs_test.go:1]: The failed cases are only for Windows? Linux is good?

Comment (devhaozi) [reply rotatelogs_test.go:1]: > The failed cases are only for Windows? Linux is good?

Not tested, I haven't got Linux development environment current.

Comment (hwbrzzl) [reply rotatelogs_test.go:1]: I think we'd better optimize the GitHub actions and go.mod first, don't change and remove the unnecessary code. It's risky to delete the test cases directly ~

Comment (devhaozi) [reply rotatelogs_test.go:1]: > I think we'd better optimize the GitHub actions and go.mod first, don't change and remove the unnecessary code. It's risky to delete the test cases directly ~

Remove Windows test and fixed some lint errors

**Review by hwbrzzl — APPROVED** [review]
Perfect 👍

**Pull request merged**